packages/compiler-core/src/transforms/transformExpression.ts

```diff
@@ -228,10 +228,10 @@ export function processExpression(
     const isAllowedGlobal = isGloballyAllowed(rawExp)
     const isLiteral = isLiteralWhitelisted(rawExp)
     if (
-      !asParams &&
-      !isScopeVarReference &&
-      !isLiteral &&
-      (!isAllowedGlobal || bindingMetadata[rawExp])
+      asParams ||
+      isScopeVarReference ||
+      isLiteral ||
+      (isAllowedGlobal && !bindingMetadata[rawExp])
```

Comment:
This isn't equivalent to the original. It looks like you're trying to apply De Morgan's law, but it's missing a `!(...)` around the whole expression.

Many of the unit tests are failing for this branch.

Reply:
tks
re-review pls

Reply:
where get  those unit tests failing results...
I cant find it

Reply:
close and reopen in order to get unit tests results.

Reply:
> where get those unit tests failing results...
> I cant find it

I checked out your branch locally and ran the tests myself.